### PR TITLE
Update repo to work with Terraform 0.13.x

### PR DESCRIPTION
**Pull request by robmorgan**
1. Update CircleCI to use Terraform 0.13.x.
1. Upgraded the Google Providers to 3.43.0.
1. Update the version constraint on all examples to require at least 0.12.26.
1. Update pre-commit hooks to add `gofmt`.
1. Drop Go Dep in favor of Go modules.
1. Upgrade the tests to support Terratest v0.30.23

In general, TF 0.13.x seems backwards compatible from a testing perspective: I did not need to change anything in `modules` in this PR.

### CODE REVIEW

**Comment by robmorgan** [comment]
Okay builds passed in https://app.circleci.com/pipelines/github/gruntwork-io/terraform-google-load-balancer/62/workflows/1c7b6469-7ddc-45c7-956c-4f132bd92606/jobs/93, so I'll mark this as ready for review.

File: .circleci/config.yml [hunk 1]
```diff
@@ -82,22 +73,35 @@ jobs:
...
     jobs:
-      - build
+      # We have to explicitly tell CircleCI to run on all tags and branches, or tag commits/pushes will not trigger
+      # builds: https://circleci.com/docs/2.0/workflows/#git-tag-job-execution.
+      - precommit:
+          filters:
+            branches:
+              only: /.*/
+            tags:
+              only: /.*/
```
Comment (rhoboat) [inline .circleci/config.yml:101]: Doesn't this need the `context: - Gruntwork Admin`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -82,22 +73,35 @@ jobs:
...
       - test:
           requires:
-            - build
-
+            - precommit
+          filters:
+            tags:
+              only: /^v.*/
```
Comment (rhoboat) [inline .circleci/config.yml:107]: I think this `test` job might also need the `context: - Gruntwork Admin`

**Review by rhoboat — COMMENTED** [review]
Left some comments about context updates.

File: .circleci/config.yml (hunk 1 above)
Comment (robmorgan) [reply .circleci/config.yml:101]: Hi @rhoboat yes good catch! We obviously missed this repo and probably several other GCP ones on the auto-run. I'll make the updates manually now.

File: .circleci/config.yml (hunk 2 above)
Comment (robmorgan) [reply .circleci/config.yml:107]: ok fixed!

**Comment by robmorgan** [comment]
ok @rhoboat thanks for the review. I've made the changes and will need a re-review when you have a moment!

**Review by rhoboat — APPROVED** [review]
LGTM!

**Pull request merged**